datamodel_code_generator/formatter/base.py

```diff
@@ -0,0 +1,184 @@
+        )
+
+    formatter_class = imported_module_.__getattribute__(import_.import_)
+
+    if not issubclass(formatter_class, BaseCodeFormatter):
+        raise TypeError(
+            f'The custom module `{custom_formatter_import}` must inherit from '
+            '`datamodel-code-generator.formatter.BaseCodeFormatter`'
+        )
+
+    custom_formatter_kwargs = custom_formatters_kwargs.get(
+        formatter_class.formatter_name, {}
+    )
+
+    return formatter_class(formatter_kwargs=custom_formatter_kwargs)
+
+
+class CodeFormattersRunner:
+    """Runner of code formatters."""
+
+    disable_default_formatter: bool
+    default_formatters: List[BaseCodeFormatter]
+    custom_formatters: List[BaseCodeFormatter]
+    custom_formatters_kwargs: Dict[str, Any]
+
+    _mapping_from_formatter_name_to_formatter_module: Dict[str, str] = {
+        'black': 'datamodel_code_generator.formatter.BlackCodeFormatter',
+        'isort': 'datamodel_code_generator.formatter.IsortCodeFormatter',
+        'ruff': 'datamodel_code_generator.formatter.RuffCodeFormatter',
+    }
+    _default_formatters: List[str] = [
+        'datamodel_code_generator.formatter.RuffCodeFormatter'
+    ]
+
+    def __init__(
+        self,
+        disable_default_formatter: bool = False,
+        default_formatter: Optional[List[str]] = None,
+        custom_formatters: Optional[List[str]] = None,
+        custom_formatters_kwargs: Optional[Dict[str, Any]] = None,
+        settings_path: Optional[Path] = None,
+        wrap_string_literal: Optional[bool] = None,
+        skip_string_normalization: bool = True,
+        known_third_party: Optional[List[str]] = None,
+    ) -> None:
+        self.disable_default_formatter = disable_default_formatter
+        self.custom_formatters_kwargs = custom_formatters_kwargs or {}
+
+        self.default_formatters = self._check_default_formatters(default_formatter)
+        self.custom_formatters = self._check_custom_formatters(custom_formatters)
+
+        self.custom_formatters_kwargs['black'] = {
+            'settings_path': settings_path,
+            'wrap_string_literal': wrap_string_literal,
+            'skip_string_normalization': skip_string_normalization,
+        }
+        self.custom_formatters_kwargs['isort'] = {
+            'settings_path': settings_path,
+            'known_third_party': known_third_party,
+        }
+
+    def _load_formatters(self, formatters: List[str]) -> List[BaseCodeFormatter]:
+        return [
+            load_code_formatter(custom_formatter_import, self.custom_formatters_kwargs)
+            for custom_formatter_import in formatters
+        ]
+
+    def _check_default_formatters(
+        self,
+        default_formatters: Optional[List[str]],
+    ) -> List[BaseCodeFormatter]:
+        if self.disable_default_formatter is True:
+            return []
+
+        if default_formatters is None:
+            return self._load_formatters(self._default_formatters)
```

Comment:
Why does the method expect the path string? 
I prefer `custom_formatter: BaseCodeFormatter`  to raw str.
We can use type-checking.

If we want to pass the custom formatted class from CLI, how about loading the custom formatted class from str only for that?
I feel that passing raw `str` is a bit confusing, since it is also intended to be used as a module by the user.

Reply:
It seems I didn't fully understand your proposal. Can you explain or give an example?

Reply:
@denisart 
I'm sorry I show you the code.
I expect the arguments.
```python
class CodeFormattersRunner:
    ...
    def __init__(
        self,
        disable_default_formatter: bool = False,
        default_formatter: Optional[List[BaseCodeFormatter]] = None,
        custom_formatters: Optional[List[BaseCodeFormatter]] = None,
```

if we give the external formatter to `CodeFormattersRunner`, we should get the `BaseCodeFormatter` by using 
 `load_code_formatter`
 
 ```python
custom_comatter = load_code_formatter("my_package.my_sub_package.FormatterName")
runner = CodeFormattersRunner(custom_formatters=[custom_comatter])
 ```
I don't think `CodeFormattersRunner` must know the full path string; `CodeFormattersRunner` only needs to handle the CodeFormatter class. What do you think?

Reply:
@koxudaxi ,
I understood your idea. Do you propose to move the loading of formatters from `CodeFormattersRunner`? This is good from an architectural point of view.

Do you propose loading formatters by path in `generate` method from `datamodel_code_generator/__init__.py`?

Reply:
@denisart 
> I understood your idea. Do you propose to move the loading of formatters from CodeFormattersRunner? This is good from an architectural point of view.

Yes!!

> Do you propose loading formatters by path in generate method from datamodel_code_generator/__init__.py?

Yes, it is. Thank you for saying what I was trying to say.

Reply:
Okay, I agree with you. I'll create a fix.